api/src/opentrons/calibration_storage/ot2/tip_length.py

```diff
@@ -35,17 +36,21 @@ def _conver_tip_length_model_to_dict(
 
 def tip_lengths_for_pipette(
     pipette_id: str,
-) -> typing.Dict[str, v1.TipLengthModel]:
+) -> typing.Dict[LabwareUri, v1.TipLengthModel]:
     tip_lengths = {}
     try:
         tip_length_filepath = config.get_tip_length_cal_path() / f"{pipette_id}.json"
         all_tip_lengths_for_pipette = io.read_cal_file(tip_length_filepath)
-        for tiprack, data in all_tip_lengths_for_pipette.items():
+        for tiprack_identifier, data in all_tip_lengths_for_pipette.items():
+            # check if tiprack hash
```

Comment:
Mind if we elaborate this comment a bit?

I'm thinking something like this.

```suggestion
            # We normally key these calibrations by their tip rack URI,
            # but older software had them keyed by their tip rack hash.
            # Migrate from the old format, if necessary.
```